### DISCUSSION TITLE
Connection string for JDBC SQLite Solaris 5.11 - Nothing works

### DISCUSSION THREAD

**Discussion by rudysack** [xerial/sqlite-jdbc#887] [Q&A] [answered]
I have a small sample program attempting to connect to our existing SQLite DB (version 3.35.5) on Solaris 5.11.  I'm using sqlite-jdbc-3.41.2.1.jar.  I've used every connection string I saw documented, including the "jdbc:sqlite::memory:" one, and I always just get "Error opening connection".   Is there anyway to turn up tracing or debug what the problem is?

**Comment by gotson** [comment]
We don't ship the native lib for Solaris, you would need to compile it yourself.

**Reply by rudysack** [reply]
1) In one case, I do get another message:  
tpsfuvvhlap004:bin:(217) > ./testSQLite.ksh
args[0]=jdbc:sqlite:tapstatus/tapstatus.db
path to 'tapstatus/tapstatus.db': '/home1/javaadm/bin/tapstatus' does not exist

This error is valid, but does it indicate that if I find the right connecting string, I still might be able to connect to this SQLite DB via Java?

2) We have two SQLite DB's on this Solaris 5.11 box that can be queried OK.  Here I query the one I'm trying to connect to via Java:
tpsfuvvhlap004: more showStatus.sh
#!/bin/sh

sqlite3 tapstatus.db <<END_SQL
.timeout 2000
.mode list
SELECT * FROM status;
END_SQL
tpsfuvvhlap004: showStatus.sh|more
2020-07-08 14:30|abrdmdtapsrv08|tsk006|Step7|NJ36|taskmate|ET06|SUBMIT/EXEC FAIL
ED
2020-07-09 10:30|hades|tsk008|enhcdny|JT|taskmate|ET06|SUBMIT/EXEC FAILED
2020-07-09 10:30|hades|tsk008|enhcdny|JY|taskmate|ET06|SUBMIT/EXEC FAILED
2020-07-09 10:30|abrdmdtapsrv08|tsk006|Step7|NJ35|taskmate|ET06|SUBMIT/EXEC FAIL
ED

tpsfuvvhlap004: uname -a
SunOS tpsfuvvhlap004 5.11 11.4.51.132.1 sun4v sparc sun4v logical-domain
tpsfuvvhlap004:

3) I'm hoping to be able to use Java to connect to this SQLite DB.  The Unix System Admin (SA) who installed SQLite left 18 months ago and hasn't been replaced.  I'm no SA, just a programmer.

Thank you.

**Reply by gotson** [reply, marked as answer]
Our library doesn't ship with the Solaris native lib for sqlite. It won't work if you don't compile it and provide it yourself.

**Comment by rudysack** [comment]
OK.  Thank you!
